### DISCUSSION TITLE
Issue with Neumann boundary conditions on .inp mesh created with gmsh

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#18499] [Q&A Meshing] [answered]
Hi, I have a mesh created with gmsh with a pretty simple geometry, but I'm having trouble using it with Neumann BCs. 

https://easyupload.io/6p4kka

The above is the raw/unedited .inp export from gmsh- when I used this I got the error:

`"Expected boundary element of dimension 2 but got 1"`

So I manually deleted all the line and surface elements, and left only the volume element. I applied Dirichlet conditions on one of the "physical groups" I created, which is a surface of one side, and it ran fine.

The problem I'm facing now is applying Neumann condition to the same surface- I get:

```
*** ERROR ***
    the following side set ids do not exist on the mesh: 3

    MOOSE distinguishes between "node sets" and "side sets" depending on whether
    you are using "Nodal" or "Integrated" BCs respectively. Node sets corresponding
    to your side sets are constructed for you by default.

    Try setting "Mesh/construct_side_list_from_node_list=true" if you see this error.
    Note: If you are running with adaptivity you should prefer using side sets.
```

So I added that argument to the Mesh block, and nothing changed. I also thought it could be the element sets created by gmsh, and since I deleted all the surface elements, I should delete the surface elsets too (and leave only the nodesets). Doing that gave me the error:

```
*** ERROR ***
Invalid dimension dim = 4
```

I also tried remaking the mesh and instead of defining the surface as the boundary, I'd define the lines that make up the surface as the boundary (in the attached file, these are named "left_tract" and "right_tract"). I did that and tried in MOOSE but just got a segmentation fault with no backtrace.

**Comment by dschwen** [comment]
@richmondodufisan you can attach files here directly without resorting to sketchy looking download sites...

See here: [dike_mesh.inp.gz](https://github.com/idaholab/moose/files/6932413/dike_mesh.inp.gz)

It would be more useful to see you gmsh script that generated that file.

**Reply by dschwen** [reply]
Meaning the .geo file! 

You can create sidesets with the `SideSetsFromBoundingBoxGenerator`, `SideSetsAroundSubdomainGenerator`, `SideSetsFromNormalsGenerator`, `ParsedGenerateSideset` etc. All those are MeshGenerators that would be chained after your `FileMeshGenerator`.

**Reply by richmondodufisan** [reply]
Hi Daniel, thanks for the tip! Also, I used the GUI to create the mesh, I guess I'll need to learn how to generate a mesh from a script so I can share that

**Comment by ngrilli** [comment, marked as answer]
Dear @richmondodufisan 

I paste below a simple 3D .geo file with definitions of physical surface.
This was working for me some time ago and I could define the boundary conditions in the moose input file
by referring to the names of the physical surfaces, such as "bottom", "top", et cetera.

You can compare with your .geo file and see if there are differences. 
Or maybe the problem is your gmsh version, not all of versions generate the same .msh files.

Hope this helps.

Best Regards,
Nicolò

```
cl__1 = 0.1;
Point(1) = {0, 0, 0, 0.1};
Point(2) = {3, 0, 0, 0.1};
Point(3) = {3, 3, 0, 0.1};
Point(4) = {0, 3, 0, 0.1};
Point(5) = {1.5, 1.5, 0, 0.1};
Point(6) = {2, 1.5, 0, 0.1};
Point(7) = {1.5, 2, 0, 0.1};
Point(8) = {1, 1.5, 0, 0.1};
Point(9) = {1.5, 1, 0, 0.1};
Point(10) = {1.5, 0, 0, 0.1};
Point(11) = {3, 1.5, 0, 0.1};
Point(12) = {1.5, 3, 0, 0.1};
Point(13) = {0, 1.5, 0, 0.1};
Line(1) = {1, 10};
Transfinite Line {1} = 16Using Progression 1;
Line(2) = {10, 2};
Transfinite Line {2} = 16Using Progression 1;
Line(3) = {2, 11};
Transfinite Line {3} = 16Using Progression 1;
Line(4) = {11, 3};
Transfinite Line {4} = 16Using Progression 1;
Line(5) = {3, 12};
Transfinite Line {5} = 16Using Progression 1;
Line(6) = {12, 4};
Transfinite Line {6} = 16Using Progression 1;
Line(7) = {4, 13};
Transfinite Line {7} = 16Using Progression 1;
Line(8) = {13, 1};
Transfinite Line {8} = 16Using Progression 1;
Line(9) = {10, 9};
Transfinite Line {9} = 11Using Progression 1;
Line(10) = {11, 6};
Transfinite Line {10} = 11Using Progression 1;
Line(11) = {12, 7};
Transfinite Line {11} = 11Using Progression 1;
Line(12) = {13, 8};
Transfinite Line {12} = 11Using Progression 1;
Circle(13) = {8, 5, 9};
Transfinite Line {13} = 11Using Progression 1;
Circle(14) = {9, 5, 6};
Transfinite Line {14} = 11Using Progression 1;
Circle(15) = {6, 5, 7};
Transfinite Line {15} = 11Using Progression 1;
Circle(16) = {7, 5, 8};
Transfinite Line {16} = 11Using Progression 1;
Line Loop(18) = {8, 1, 9, -13, -12};
Plane Surface(18) = {18};
Recombine Surface {18};
Line Loop(19) = {13, 14, 15, 16};
Plane Surface(19) = {19};
Recombine Surface {19};
Line Loop(20) = {9, 14, -10, -3, -2};
Plane Surface(20) = {20};
Recombine Surface {20};
Line Loop(22) = {10, 15, -11, -5, -4};
Plane Surface(22) = {22};
Recombine Surface {22};
Line Loop(24) = {11, 16, -12, -7, -6};
Plane Surface(24) = {24};
Recombine Surface {24};
Extrude {0, 0, 0.1} {
  Surface{18, 19, 20, 22, 24};
  Layers{1};
  Recombine;
}
Physical Surface("bottom") = {38, 99};
Physical Surface("top") = {122, 153};
Physical Surface("right") = {95, 126};
Physical Surface("left") = {149, 34};
Physical Surface("front") = {154, 127, 100, 51};
Physical Surface("back") = {18, 20, 22, 24};
Physical Volume("all") = {1, 2, 3, 4, 5};
Physical Volume("pore") = {2};
Physical Volume("allminuspore") = {1, 3, 4, 5};
```

**Reply by richmondodufisan** [reply]
So I have this:


cl__1 = 1;
Point(1) = {0, 0, 0, cl__1};
Point(2) = {300, 0, 0, cl__1};
Point(3) = {136, 80, 0, cl__1};
Point(4) = {164, 80, 0, cl__1};
Point(5) = {164, 80, 25, cl__1};
Point(6) = {136, 80, 25, cl__1};
Point(7) = {0, 0, 25, cl__1};
Point(8) = {300, 0, 25, cl__1};
Line(1) = {7, 6};
Line(2) = {6, 5};
Line(3) = {5, 8};
Line(4) = {8, 2};
Line(5) = {2, 1};
Line(6) = {1, 7};
Line(7) = {1, 3};
Line(8) = {3, 6};
Line(9) = {3, 4};
Line(10) = {4, 5};
Line(11) = {4, 2};
Line(12) = {8, 7};
Curve Loop(1) = {1, 2, 3, 12};
Plane Surface(1) = {1};
Curve Loop(2) = {7, 8, -1, -6};
Plane Surface(2) = {2};
Curve Loop(3) = {2, -10, -9, 8};
Plane Surface(3) = {3};
Curve Loop(4) = {3, 4, -11, 10};
Plane Surface(4) = {4};
Curve Loop(5) = {12, -6, -5, -4};
Plane Surface(5) = {5};
Curve Loop(6) = {11, 5, 7, 9};
Plane Surface(6) = {6};
Surface Loop(1) = {1, 2, 6, 4, 5, 3};
Volume(1) = {1};
Physical Surface("front") = {1};
Physical Surface("back") = {6};
Physical Surface("left") = {2};
Physical Surface("right") = {4};
Physical Surface("top") = {3};
Physical Surface("bottom") = {5};


which seems slightly different from yours. In my version lines are referred to as curves. There's also a "recombine surface" in yours that I can't find- what version of gmsh are you using? Mine is 4.8.4.

**Reply by richmondodufisan** [reply]
I also tried giving MOOSE .msh files instead of .inp files, and I got this error:

```
Error: Exodus requires all elements with a given subdomain ID to be the same type.
Can't write both PYRAMID5 and TET4 in the same block!Error: Exodus requires all elements with a given subdomain ID to be the same type.
Can't write both PYRAMID5 and TET4 in the same block!
```

But in my mesh preferences, I selected:
2D Algorithm: MeshAdapt (I also tried "Automatic" and "Frontal Delaunay for Quads")
3D Algorithm: Delaunay (I tried all non "experimental" algorithms available).
2D Recombination Algorithm: Simple Full Quad
Subdivision Algorithm: All Quads
